exporter/datadogexporter/config.go

```diff
@@ -0,0 +1,549 @@
+	Metrics MetricsConfig `mapstructure:"metrics"`
+
+	// Traces defines the Traces exporter specific configuration
+	Traces TracesConfig `mapstructure:"traces"`
+
+	// HostMetadata defines the host metadata specific configuration
+	HostMetadata HostMetadataConfig `mapstructure:"host_metadata"`
+
+	// SendMetadata defines whether to send host metadata
+	// This is undocumented and only used for unit testing.
+	//
+	// This can't be disabled if `only_metadata` is true.
+	// Deprecated: [v0.49.0] Use `host_metadata::enabled` (HostMetadata.Enabled) instead.
+	SendMetadata bool `mapstructure:"send_metadata"`
+
+	// OnlyMetadata defines whether to only send metadata
+	// This is useful for agent-collector setups, so that
+	// metadata about a host is sent to the backend even
+	// when telemetry data is reported via a different host.
+	//
+	// This flag is incompatible with disabling host metadata,
+	// `use_resource_metadata`, or `host_metadata::hostname_source != first_resource`
+	OnlyMetadata bool `mapstructure:"only_metadata"`
+
+	// UseResourceMetadata defines whether to use resource attributes
+	// for completing host metadata (such as the hostname or host tags).
+	//
+	// By default this is true: the first resource attribute getting to
+	// the exporter will be used for host metadata.
+	// Disable this in the Collector if you are using an agent-collector setup.
+	// Deprecated: [v0.49.0] Use `host_metadata::hostname_source` (HostMetadata.HostnameSource) instead.
+	UseResourceMetadata bool `mapstructure:"use_resource_metadata"`
+
+	// warnings stores non-fatal configuration errors.
+	warnings []error
+}
+
+// logWarnings logs warning messages that were generated on unmarshaling.
+func (c *Config) logWarnings(logger *zap.Logger) {
+	for _, err := range c.warnings {
+		logger.Warn(fmt.Sprintf("%v", err))
+	}
+}
+
+func (c *Config) Validate() error {
+	if c.OnlyMetadata && (!c.HostMetadata.Enabled || c.HostMetadata.HostnameSource != HostnameSourceFirstResource) {
+		return errNoMetadata
+	}
+
+	if err := valid.Hostname(c.Hostname); c.Hostname != "" && err != nil {
+		return fmt.Errorf("hostname field is invalid: %w", err)
+	}
+
+	if c.API.Key == "" {
+		return errUnsetAPIKey
+	}
+
+	if c.Traces.IgnoreResources != nil {
+		for _, entry := range c.Traces.IgnoreResources {
+			_, err := regexp.Compile(entry)
+			if err != nil {
+				return fmt.Errorf("'%s' is not valid resource filter regular expression", entry)
+			}
+		}
+	}
+
+	if c.Traces.SpanNameRemappings != nil {
+		for key, value := range c.Traces.SpanNameRemappings {
+			if value == "" {
+				return fmt.Errorf("'%s' is not valid value for span name remapping", value)
+			}
+			if key == "" {
+				return fmt.Errorf("'%s' is not valid key for span name remapping", key)
+			}
+		}
+	}
+
+	err := c.Metrics.HistConfig.validate()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Config) Unmarshal(configMap *confmap.Conf) error {
```

Comment:
What is this method? What is its purpose? Why is it here? Can we document it?

Reply:
It implements this interface: https://pkg.go.dev/go.opentelemetry.io/collector/config#Unmarshallable I added a comment and an interface assertion